src/system_metrics/mongoose_system_metrics_collector.erl

```diff
@@ -38,7 +38,9 @@ report_getters() ->
     ].
 
 get_hosts_count() ->
-    Hosts = ejabberd_config:get_global_option(hosts),
+    HostTypes = ejabberd_config:get_global_option_or_default(host_types, []),
+    Domains = lists:flatten([mongoose_domain_core:get_domains_by_host_type(HT) || HT <- HostTypes]),
+    Hosts = ejabberd_config:get_global_option_or_default(hosts, []) ++ Domains,
```

Comment:
Please add a new function to `mongoose_domain_core` to get all domains by simply reading the whole ETS table. Then, call this function from here.